CI: matplotlib failures

https://travis-ci.org/pandas-dev/pandas/jobs/542921706#L2423

```
=================================== FAILURES ===================================
_______________________ TestDataFramePlots.test_subplots _______________________
[gw1] linux -- Python 3.6.8 /home/travis/miniconda3/envs/pandas-dev/bin/python
self = <pandas.tests.plotting.test_frame.TestDataFramePlots object at 0x7f507ed05128>
    @pytest.mark.slow
    def test_subplots(self):
        df = DataFrame(np.random.rand(10, 3),
                       index=list(string.ascii_letters[:10]))
    
        for kind in ['bar', 'barh', 'line', 'area']:
            axes = df.plot(kind=kind, subplots=True, sharex=True, legend=True)
            self._check_axes_shape(axes, axes_num=3, layout=(3, 1))
            assert axes.shape == (3, )
    
            for ax, column in zip(axes, df.columns):
                self._check_legend_labels(ax,
                                          labels=[pprint_thing(column)])
    
            for ax in axes[:-2]:
                self._check_visible(ax.xaxis)  # xaxis must be visible for grid
                self._check_visible(ax.get_xticklabels(), visible=False)
                self._check_visible(
>                   ax.get_xticklabels(minor=True), visible=False)
pandas/tests/plotting/test_frame.py:387: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pandas.tests.plotting.test_frame.TestDataFramePlots object at 0x7f507ed05128>
collections = <a list of 2 Text xticklabel objects>, visible = False
    def _check_visible(self, collections, visible=True):
        """
        Check each artist is visible or not
    
        Parameters
        ----------
        collections : matplotlib Artist or its list-like
            target Artist or its list or collection
        visible : bool
            expected visibility
        """
        from matplotlib.collections import Collection
        if not isinstance(collections,
                          Collection) and not is_list_like(collections):
            collections = [collections]
    
        for patch in collections:
>           assert patch.get_visible() == visible
E           AssertionError
pandas/tests/plotting/common.py:137: AssertionError
```


I'm looking into it now, but will have to go in 30 minutes.

--- a/ci/deps/travis-36-cov.yaml
+++ b/ci/deps/travis-36-cov.yaml
@@ -20,7 +20,8 @@ dependencies:
   # https://github.com/pydata/pandas-gbq/issues/271
   - google-cloud-bigquery<=1.11
   - psycopg2
-  - pyarrow=0.9.0
+  # pyarrow segfaults on load: https://github.com/pandas-dev/pandas/issues/26716
+  # - pyarrow=0.9.0
   - pymysql
   - pytables
   - python-snappy

--- a/pandas/tests/plotting/common.py
+++ b/pandas/tests/plotting/common.py
@@ -441,9 +441,9 @@ class TestPlotBase:
         import matplotlib as mpl
 
         def is_grid_on():
-            xoff = all(not g.gridOn
+            xoff = all(not g.gridline.get_visible()
                        for g in self.plt.gca().xaxis.get_major_ticks())
-            yoff = all(not g.gridOn
+            yoff = all(not g.gridline.get_visible()
                        for g in self.plt.gca().yaxis.get_major_ticks())
             return not (xoff and yoff)
 

--- a/pandas/tests/plotting/common.py
+++ b/pandas/tests/plotting/common.py
@@ -36,6 +36,7 @@ class TestPlotBase:
 
         self.mpl_ge_2_2_3 = plotting._compat._mpl_ge_2_2_3()
         self.mpl_ge_3_0_0 = plotting._compat._mpl_ge_3_0_0()
+        self.mpl_ge_3_1_0 = plotting._compat._mpl_ge_3_1_0()
 
         self.bp_n_objects = 7
         self.polycollection_factor = 2

--- a/pandas/tests/plotting/common.py
+++ b/pandas/tests/plotting/common.py
@@ -441,10 +441,18 @@ class TestPlotBase:
         import matplotlib as mpl
 
         def is_grid_on():
-            xoff = all(not g.gridline.get_visible()
-                       for g in self.plt.gca().xaxis.get_major_ticks())
-            yoff = all(not g.gridline.get_visible()
-                       for g in self.plt.gca().yaxis.get_major_ticks())
+            xticks = self.plt.gca().xaxis.get_major_ticks()
+            yticks = self.plt.gca().yaxis.get_major_ticks()
+            # for mpl 2.2.2, gridOn and gridline.get_visible disagree.
+            # for new MPL, they are the same.
+
+            if self.mpl_ge_3_0_0:
+                xoff = all(not g.gridline.get_visible() for g in xticks)
+                yoff = all(not g.gridline.get_visible() for g in yticks)
+            else:
+                xoff = all(not g.gridOn for g in xticks)
+                yoff = all(not g.gridOn for g in yticks)
+
             return not (xoff and yoff)
 
         spndx = 1

--- a/pandas/tests/plotting/test_frame.py
+++ b/pandas/tests/plotting/test_frame.py
@@ -383,8 +383,10 @@ class TestDataFramePlots(TestPlotBase):
             for ax in axes[:-2]:
                 self._check_visible(ax.xaxis)  # xaxis must be visible for grid
                 self._check_visible(ax.get_xticklabels(), visible=False)
-                self._check_visible(
-                    ax.get_xticklabels(minor=True), visible=False)
+                if not (kind == 'bar' and self.mpl_ge_3_1_0):
+                    # change https://github.com/pandas-dev/pandas/issues/26714
+                    self._check_visible(
+                        ax.get_xticklabels(minor=True), visible=False)
                 self._check_visible(ax.xaxis.get_label(), visible=False)
                 self._check_visible(ax.get_yticklabels())
 

